cedar-policy-core/src/entities/json/schema_types.rs

```diff
@@ -353,12 +353,8 @@ impl Diagnostic for NontrivialResidualError {
 /// For records, we can't know whether the attributes in the given record are
 /// required or optional.
 /// This function, when given a record that has keys A, B, and C, will return a
-/// `SchemaType` where A, B, and C are all marked as optional attributes, but no
+/// `SchemaType` where A, B, and C are all marked as required attributes, but no
 /// other attributes are possible.
-/// That is, this assumes that all existing attributes are optional, but that no
-/// other optional attributes are possible.
-/// Compared to marking A, B, and C as required, this allows the returned
-/// `SchemaType` to `is_consistent_with()` more types.
```

Comment:
Presumably, now with this change, `SchemaType` will be `is_consistent_with()` less types (unless this comment was wrong).  Are we reasonably sure this won't cause other problems?

Reply:
If `is_consistent_with` is just a typing judgement, then this should be fine because `required <: optional`.

Reply:
`is_consistent_with` doesn't use the validator's type system. Let's double check that it does the right thing.

Reply:
No longer modifying `is_consistent_with`.